ros2node/ros2node/api/__init__.py

```diff
@@ -29,16 +29,18 @@ def get_absolute_node_name(node_name):
     return node_name
 
 
-def parse_node_name(full_node_name):
-    tokens = full_node_name.split('/')
-    if 1 > len(tokens):
-        raise RuntimeError('Invalid node name: ' + full_node_name)
-    node_name = full_node_name
-    namespace = '/'
+def parse_node_name(node_name):
+    if (node_name[0] == '/'):
+        node_name = node_name[1:]
+    tokens = node_name.split('/')
```

Comment:
Since the following code only uses two parts of this I would suggest splitting explicitly for that: `rsplit('/', 1)`

With this change it might be better to prepend `/` on demand to the parameter rather then removing it if present. Just an idea.